### PR TITLE
Fix the tuts broken by #810, reform syntax

### PR DESCRIPTION
**Pull request by rossabaker**
Some tuts broke because of failure to deal with `MaybeResponse`.  We had an `orNotFound` in our specs which also turns out to be handy in the REPL, so I introduced syntax.

As we aim toward cats as the primary foundation, it makes sense to use a cats-like organization.  So I fell down a rabbit hole.

### CODE REVIEW

**Comment by reactormonk** [comment]
I'm kind of on the fence of `Successful(resp)`, took me a while to figure out it's not actually an Either is some way.

**Comment by rossabaker** [comment]
I'm always annoyed how that one is one import deeper than most other things.  Maybe we'd both be happier if we unnest it one level and could reference it as `Status.Successful`?

**Comment by ChristopherDavenport** [comment]
Never a bad things to make your imports more readable.

**Comment by rossabaker** [comment]
ab14005 implements my proposal above, but while fixing the tut, I stubbed my toe on this again:

```scala
scala> import Status._, dsl._
import Status._
import dsl._

scala> NotFound()
res2: scalaz.concurrent.Task[org.http4s.Response] = scalaz.concurrent.Task@70ec248f

scala> import dsl._, Status._
import dsl._
import Status._

scala> NotFound()
<console>:40: error: org.http4s.Status does not take parameters
       NotFound()
```

`import Status.ResponseClass._` was fine with the DSL.  `import Status._` causes unexpected trouble.  This was true before, but people were less likely to do it.

**Comment by rossabaker** [comment]
#774 claims another build victim.

File: docs/src/main/tut/middleware.md [hunk 1]
```diff
@@ -33,10 +33,10 @@ the wrapped service like this.
 
 ```tut:book
 def myMiddle(service: HttpService, header: Header): HttpService = Service.lift { req =>
-  service(req).map {resp =>
-    if (resp.status.isSuccess)
+  service(req) map {
```
Comment (bryce-anderson) [inline docs/src/main/tut/middleware.md]: We're preferring infix syntax now? This is really going to mess me up between work where we're trying to get rid of it and here. :smile:

**Pull request merged**